core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -617,9 +617,9 @@ class GroupCoordinator(val brokerId: Int,
                        leavingMembers: List[MemberIdentity],
                        responseCallback: LeaveGroupResult => Unit): Unit = {
 
-    def removeCurrentMemberFromGroup(group: GroupMetadata, memberId: String): Unit = {
+    def removeCurrentMemberFromGroup(group: GroupMetadata, memberId: String, leaveReason: String): Unit = {
```

Comment:
Should we use `Option[String]` for the `leaveReason` and handle the case where it is not provided?